src/storage-controller/src/lib.rs

```diff
@@ -837,6 +837,12 @@ where
                                 self.partially_truncate_statement_log().await;
                             }
                         }
+
+                        // Truncate compute-maintained collections.
+                        IntrospectionType::ComputeDependencies
+                        | IntrospectionType::ComputeReplicaHeartbeats => {
+                            self.reconcile_managed_collection(id, vec![]).await;
```

Comment:
this brings the entire collection into memory; is that what happened in the past when it was adapter-managed? if so, seems fine

Reply:
> is that what happened in the past when it was adapter-managed

yes

Reply:
To elaborate on Brennan's correct but brief statement: These relations were system tables before, and [all system tables get truncated during bootstrap](https://github.com/teskje/materialize/blob/4fea5f8386cb56d299d9674043f67e1a7e788f3d/src/adapter/src/coord.rs#L1704-L1731). They are stash collections, which are differential collections, so truncation works the same as for persist collections, by reading the contents into memory and then issuing retractions for them.